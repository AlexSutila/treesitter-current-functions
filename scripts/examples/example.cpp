#include <utility>

/*
 * Test 1: Basics, method definitions both inside and outside the class definition. Also testing basic
 *          functions with primitive, pointer, and reference return types.
 */

class test_1_out {

private:
    int a, b;

public:
    std::pair<int, int> get_ab() const;
    test_1_out(int _a, int _b);
    ~test_1_out();
};
test_1_out *test_1_out_inst = nullptr;

test_1_out::test_1_out(int _a, int _b)   { a = _a; b = _b; }
test_1_out::~test_1_out()                {  }
std::pair<int, int> test_1_out::get_ab() const { return { a, b }; }

test_1_out get_test_1_out_inst_reg() {
    return *test_1_out_inst;
}
test_1_out *get_test_1_out_inst_ptr() {
    return test_1_out_inst;
}
test_1_out &get_test_1_out_inst_ref() {
   return *test_1_out_inst;
}

