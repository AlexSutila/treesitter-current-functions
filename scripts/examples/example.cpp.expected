{ {
    function_name = "test_1_out::test_1_out",
    line_number = 21
  }, {
    function_name = "test_1_out::~test_1_out",
    line_number = 22
  }, {
    function_name = "test_1_out::get_ab",
    line_number = 23
  }, {
    function_name = "get_test_1_out_inst_reg",
    line_number = 25
  }, {
    function_name = "get_test_1_out_inst_ptr",
    line_number = 28
  }, {
    function_name = "get_test_1_out_inst_ref",
    line_number = 31
  }, {
    function_name = "test_2_out<T>::test_2_out",
    line_number = 52
  }, {
    function_name = "test_2_out<T>::get_ab",
    line_number = 53
  }, {
    function_name = "get_test_2_out_inst_reg",
    line_number = 55
  }, {
    function_name = "get_test_2_out_inst_ptr",
    line_number = 58
  }, {
    function_name = "get_test_2_out_inst_ref",
    line_number = 61
  } }
